File: src/ui/Navigation.cpp
#include "Navigation.h"
#include <iostream>

void Navigation::goForward() {
    if (!forwardStack.empty()) {
        history.push(currentPage);
        currentPage = forwardStack.top();
        forwardStack.pop();
    } else {
        std::cout << "No forward history available.\n";
    }
}

void Navigation::goBack() {
    if (!history.empty()) {
        forwardStack.push(currentPage);
        currentPage = history.top();
        history.pop();
    } else {
        std::cout << "No back history available.\n";
    }
}

void Navigation::refresh() {
    std::cout << "Refreshing: " << currentPage << "\n";
}

void Navigation::goHome() {
    while (!history.empty()) history.pop();
    while (!forwardStack.empty()) forwardStack.pop();
    currentPage = "home";
    std::cout << "Navigated to home.\n";
}

void Navigation::navigateTo(const std::string& url) {
    if (!currentPage.empty()) {
        history.push(currentPage);
    }
    currentPage = url;
    while (!forwardStack.empty()) forwardStack.pop();
    std::cout << "Navigated to: " << url << "\n";
}

void Navigation::displayHistory() const {
    std::stack<std::string> temp = history;
    std::cout << "History:\n";
    while (!temp.empty()) {
        std::cout << temp.top() << "\n";
        temp.pop();
    }
}

File: src/ui/navigation.rs
use crate::ui::TabManager;

pub struct Navigation {
    tab_manager: TabManager,
}

impl Navigation {
    pub fn new(tab_manager: TabManager) -> Self {
        Navigation { tab_manager }
    }

    pub fn go_back(&mut self) {
        if let Some(tab) = self.tab_manager.get_active_tab_mut() {
            tab.webview.evaluate_script("history.back()").ok();
        }
    }

    pub fn go_forward(&mut self) {
        if let Some(tab) = self.tab_manager.get_active_tab_mut() {
            tab.webview.evaluate_script("history.forward()").ok();
        }
    }

    pub fn refresh(&mut self) {
        if let Some(tab) = self.tab_manager.get_active_tab_mut() {
            tab.webview.evaluate_script("location.reload()").ok();
        }
    }

    pub fn navigate_to(&mut self, url: &str) {
        let id = self.tab_manager.get_active_tab().map(|tab| tab.id).unwrap_or(0);
        if let Some(tab) = self.tab_manager.get_active_tab_mut() {
            tab.webview.load_url(url);
        }
        self.tab_manager.update_tab_url(id, url.to_string());
    }

    pub fn get_current_url(&self) -> Option<String> {
        self.tab_manager.get_active_tab().map(|tab| tab.url.clone())
    }
}